packages/http/fetch/src/kiotaClientFactory.ts

```diff
@@ -0,0 +1,19 @@
+import { Middleware, MiddlewareFactory } from ".";
+import { HttpClient } from "./httpClient";
+
+/**
+ * @class
+ * Class containing function(s) related to the middleware pipelines.
+ */
+export class KiotaClientFactory {
+	/**
+	 * @public
+	 * @static
+	 * Returns an instance of HttpClient with the provided middlewares and custom fetch implementation
+	 * @returns a HttpClient instance
+	 */
+	public static create(customFetch: (request: string, init: RequestInit) => Promise<Response> = fetch as any, middlewares?: Middleware[]): HttpClient {
```

Comment:
@baywet instead of the optional parameters would it be helpful if I create additional signatures or a builder, example:
```javascript
create()
createClientWithMidlewares(middleware: Middleware[]);
createClientWithCustomFetch(customFetch: (request: string, init: RequestInit) => Promise<Response>);
createClientWithCustomFetchAndMiddleware(customFetch: (request: string, init: RequestInit) => Promise<Response>, middlewares: Middleware[])
```

Reply:
I think this suggestion doesn't look like TypeScript usual approach: people expect optional parameters given there's good doc comments.
Now, maybe your suggestion is to move the middleware array to the first position? That I think would make more sense because I suspect most people will want to change only the middleware set rather than the fetch implementation under the hood.
Actually because of #1083, we might remove the custom fetch argument all together, but this will come in a latter PR I suspect.

Reply:
+1 to @baywet comment!